go-controller/pkg/factory/factory.go

```diff
@@ -276,10 +280,34 @@ func (i *informer) shutdown() {
 	}
 }
 
+func newInformerLister(oType reflect.Type, sharedInformer cache.SharedIndexInformer) (listerInterface, error) {
+	switch oType {
+	case podType:
+		return listers.NewPodLister(sharedInformer.GetIndexer()), nil
+	case serviceType:
+		return listers.NewServiceLister(sharedInformer.GetIndexer()), nil
+	case endpointsType:
+		return listers.NewEndpointsLister(sharedInformer.GetIndexer()), nil
+	case namespaceType:
+		return listers.NewNamespaceLister(sharedInformer.GetIndexer()), nil
+	case nodeType:
+		return listers.NewNodeLister(sharedInformer.GetIndexer()), nil
+	}
+	return nil, fmt.Errorf("cannot create lister from type %v", oType)
+}
+
 func newBaseInformer(oType reflect.Type, sharedInformer cache.SharedIndexInformer) *informer {
+	lister, err := newInformerLister(oType, sharedInformer)
+
+	if err != nil {
+		logrus.Errorf(err.Error())
```

Comment:
@abhat also, if we're going to get an error here, let's actually return it all the way up to the callers.